src/components/MapSelectors/GlobalSelector.js

```diff
@@ -235,10 +235,24 @@ const GlobalSelector = ({ handleChange, extendRight }) => {
     }
   };
 
+  const escapeKeyStateReducer = (state, changes) => {
```

Comment:
Can you add a comment explaining what's going on here?

Reply:
Will do!